compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -836,6 +836,10 @@ impl<'a> Parser<'a> {
             assert!(found_raw);
             let mutability = self.parse_const_or_mut().unwrap();
             (ast::BorrowKind::Raw, mutability)
+        } else if let Some((ast::Pinnedness::Pinned, mutbl)) = self.parse_pin_and_mut() {
+            // `pin [ const | mut ]`.
+            // `pin` has been gated in `self.parse_pin_and_mut()` so we don't need to gate it here.
```

Comment:
```suggestion
            // `pin [ const | mut ]`.
            //
            // `pin` has been gated in `self.parse_pin_and_mut()` so we don't
            // need to gate it here.
```

The comment at the top of the function should also be adjusted to mention `pin`.